### DISCUSSION TITLE
Build Fails with latest commit.

### DISCUSSION THREAD

**Discussion by SomerledDesign** [PlummersSoftwareLLC/NightDriverStrip#148] [Q&A] [answered]
I am trying to build the 'ledstrip' demo or any one of them, for that matter, and I get a lot of this...
Am I forgetting something?
I have the latest vscode and v1.8.19 of arduino...

The thing is I had an earlier commit of NightDriverStrip on my machine and it built well.  So I downloaded the latest and haven't been able to build since.

============== partial error log
pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:20:25: error: expected '}' before '__VA_OPT__'
 #define STUFF_P(a, ...) __VA_OPT__(a)
                         ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:19:21: note: in definition of macro 'EXPAND'
 #define EXPAND(...) __VA_ARGS__
                     ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:21:20: note: in expansion of macro 'STUFF_P'
 #define STUFF(...) STUFF_P(__VA_ARGS__)
                    ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:25:28: note: in expansion of macro 'STUFF'
 #define COND_P(cond, a, b) STUFF(a, cond)STUFF(b, NEGATE(cond))
                            ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:26:33: note: in expansion of macro 'COND_P'
 #define COND(cond, a, b) EXPAND(COND_P(cond, a, b))
                                 ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:292:5: note: in expansion of macro 'COND'
     COND(DECODE_RC5 || SEND_RC5,
     ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:20:25: error: expected ',' or ';' before '__VA_OPT__'
 #define STUFF_P(a, ...) __VA_OPT__(a)
                         ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:19:21: note: in definition of macro 'EXPAND'
 #define EXPAND(...) __VA_ARGS__
                     ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:21:20: note: in expansion of macro 'STUFF_P'
 #define STUFF(...) STUFF_P(__VA_ARGS__)
                    ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:25:28: note: in expansion of macro 'STUFF'
 #define COND_P(cond, a, b) STUFF(a, cond)STUFF(b, NEGATE(cond))
                            ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:26:33: note: in expansion of macro 'COND_P'
 #define COND(cond, a, b) EXPAND(COND_P(cond, a, b))
                                 ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:292:5: note: in expansion of macro 'COND'
     COND(DECODE_RC5 || SEND_RC5,
     ^
.pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:538:1: error: expected declaration before '}' token
 };
 ^
*** [.pio/build/ledstrip/lib601/IRremoteESP8266/IRtext.cpp.o] Error 1
========================================== [FAILED] Took 12.61 seconds ==========================================

Environment    Status    Duration
-------------  --------  ------------
ledstrip       FAILED    00:00:12.606
===================================== 1 failed, 0 succeeded in 00:00:12.606 =====================================

**Comment by davepl** [comment, marked as answer]
All of our code is built using PlatformIO under VSCode, which it does look like you’re using.

 

Try running 

 

platformio pkg update

 

See if that helps!

 

*	Dave

 

From: somerledDesign ***@***.***> 
Sent: Sunday, September 18, 2022 9:28 AM
To: PlummersSoftwareLLC/NightDriverStrip ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [PlummersSoftwareLLC/NightDriverStrip] Build Fails with latest commit. (Discussion #148)

 

I am trying to build the 'ledstrip' demo or any one for that matter and I get a lot of this...
Am I forgetting something?
I have the latest vscode and v1.8.19 of arduino...

The thing is I had an earlier commit of NightDriverStrip on my machine and it built well. So I downloaded the latest and haven't been able to build since.

pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:20:25: error: expected '}' before '__VA_OPT__' #define STUFF_P(a, ...) __VA_OPT__(a) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:19:21: note: in definition of macro 'EXPAND' #define EXPAND(...) __VA_ARGS__ ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:21:20: note: in expansion of macro 'STUFF_P' #define STUFF(...) STUFF_P(__VA_ARGS__) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:25:28: note: in expansion of macro 'STUFF' #define COND_P(cond, a, b) STUFF(a, cond)STUFF(b, NEGATE(cond)) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:26:33: note: in expansion of macro 'COND_P' #define COND(cond, a, b) EXPAND(COND_P(cond, a, b)) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:292:5: note: in expansion of macro 'COND' COND(DECODE_RC5 || SEND_RC5, ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:20:25: error: expected ',' or ';' before '__VA_OPT__' #define STUFF_P(a, ...) __VA_OPT__(a) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:19:21: note: in definition of macro 'EXPAND' #define EXPAND(...) __VA_ARGS__ ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:21:20: note: in expansion of macro 'STUFF_P' #define STUFF(...) STUFF_P(__VA_ARGS__) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:25:28: note: in expansion of macro 'STUFF' #define COND_P(cond, a, b) STUFF(a, cond)STUFF(b, NEGATE(cond)) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRmacros.h:26:33: note: in expansion of macro 'COND_P' #define COND(cond, a, b) EXPAND(COND_P(cond, a, b)) ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:292:5: note: in expansion of macro 'COND' COND(DECODE_RC5 || SEND_RC5, ^ .pio/libdeps/ledstrip/IRremoteESP8266/src/IRtext.cpp:538:1: error: expected declaration before '}' token }; ^ *** [.pio/build/ledstrip/lib601/IRremoteESP8266/IRtext.cpp.o] Error 1 ========================================== [FAILED] Took 12.61 seconds ========================================== 

Environment Status Duration

  _____  

ledstrip FAILED 00:00:12.606
===================================== 1 failed, 0 succeeded in 00:00:12.606 =====================================

**Reply by SomerledDesign** [reply]
I had the 5.1.1.3a problem... removed that. ('Mulitple Cores')
Then removed and reinstalled PlatformIO extension in VS Code.  Reboot (I'm on a mac but windows ways die hard).
Then open Platform IO Core CL and perform the above suggestion [platformio pkg update]

$ platformio pkg update
Resolving all-deps dependencies...
Already up-to-date.
Resolving demo dependencies...
Already up-to-date.
Resolving wrover-demo dependencies...
Already up-to-date.
Resolving ledstrip dependencies...
> Library Manager: Updating IRremoteESP8266 @ 2.8.3
> Library Manager: Removing IRremoteESP8266 @ 2.8.3
> Library Manager: IRremoteESP8266@2.8.3 has been removed!
> Library Manager: Installing crankyoldgit/IRremoteESP8266 @ 2.8.4
> Downloading  [####################################]  100%
> Unpacking  [####################################]  100%
> Library Manager: IRremoteESP8266@2.8.4 has been installed!
Library Manager: Updating U8g2 @ 2.33.15
Library Manager: Removing U8g2 @ 2.33.15
Library Manager: U8g2@2.33.15 has been removed!
Library Manager: Installing olikraus/U8g2 @ 2.34.0
Downloading  [####################################]  100%          
Unpacking  [####################################]  100%          
Library Manager: U8g2@2.34.0 has been installed!
Resolving generic dependencies...
Already up-to-date.
.
.
.
Resolving cube dependencies...
Already up-to-date.

=============
Now it works great.
Thanks Dave

Kevin
New Boston
